src/tests/ftest/rebuild/container_rf_enforce.py

```diff
@@ -0,0 +1,146 @@
+           for positive testcase, enable the exception with write objects
+           for negative testcase, disable the exception with write objects
+           and expecting failure with RC: -1003.
+
+        Args:
+            negative_test (Bool, optional): container write_objects to handle
+                test Exception and parse for return error code.
+        """
+        der_inval = "RC: -1003"
+        self.log.info(
+            "==>(1)Create pool and container with redundant factor,"
+            " start background IO object write")
+        self.container.create()
+        if not negative_test:
+            self.container.write_objects(
+                self.inputs.rank.value[0], self.inputs.object_class.value)
+        else:
+            try:
+                self.container.write_objects_wo_failon(
+                    self.inputs.rank.value[0], self.inputs.object_class.value)
+                self.fail("#Container redundancy factor with an invallid "
+                           "object_class traffic passed, expecting Fail")
+            except DaosTestError as error:
+                self.log.info(error)
+                if der_inval in str(error):
+                    self.log.info("==Negative test, excepting DaosApiError "
+                                  "RC: -1003 found.")
+                else:
+                    self.fail("#Negative test, container redundancy factor "
+                              "test failed, return error RC: -1003 not found")
+
+    def execute_rf_enforce_test(self, create_container=True):
+        """Execute the rebuild test steps for container rf test.
+
+        Args:
+            create_container (bool, optional): should the test create a
+                container. Defaults to True.
+        """
+        # Get the test params and var
+        self.setup_test_pool()
+        self.daos_cmd = DaosCommand(self.bin)
+        if create_container:
+            self.setup_test_container()
+        oclass = self.inputs.object_class.value
+        num_of_ranks = len(self.inputs.rank.value)
+        rf = ''.join(self.container.properties.value.split(":"))
+        rf_num = int(re.search(r"rf([0-9]+)", rf).group(1))
+        if num_of_ranks > rf_num:
+            expect_cont_status = "UNCLEAN"
+        else:
+            expect_cont_status = "HEALTHY"
+        negative_test = True
+        if "OC_SX" in oclass and rf_num < 1:
+            negative_test = False
+        elif ("OC_RP_2" in oclass and rf_num < 2) or (
+            "OC_RP_3" in oclass and rf_num < 3):
+            negative_test = False
+        # Create a pool and verify the pool information before rebuild
+        self.create_test_pool()
+        # Create a container and write objects
+        self.create_test_container_and_write_obj(negative_test)
+        if negative_test:
+            self.log.info("Negative test passed")
+        else:
+        # Verify the rank to be excluded has at least one object
```

Comment:
```suggestion
            # Verify the rank to be excluded has at least one object
```
Isn't this supposed to be indented?

Reply:
Please see the update. Thanks.